lib/matplotlib/afm.py

```diff
@@ -91,7 +91,7 @@ def _sanity_check(fh):
     # do something else with the file.
     pos = fh.tell()
     try:
-        line = fh.readline()
+        line = next(fh)
```

Comment:
The problem is that the outer loop may be using `for line in fh` first before passing the handle to the function at some point, and Py2 (not Py3) doesn't allow (raises an exception on) mixing `for line in fh` and `fh.readline` due to the fact that the former uses a readahead buffer (https://docs.python.org/2/library/stdtypes.html#file.next).